### PR TITLE
Added warning about v2 version string encoding

### PR DESCRIPTION
**Pull request by daidoji**
Added warning about b64 encoding of version2 version string.

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -1220,6 +1220,21 @@ The first four characters, `PPPP` indicate the protocol. Each genus of a given C
...
+::: warning non-canonical base64
+This is a non-canonical encoding using Base64 indicies. Most libraries will drop bits that aren't on a byte boundary using many rfc4648 compliant libraries if you just call decode on these characters naively.
+
+example) in python (with padding character for demonstration) our version 2.00 maps to "CAA" as above.
+```python
+>>> base64.urlsafe_b64decode("CAA=")
+b'\x08\x00'
+```
+
+Which is two bytes.  However, three base64 characters in this version scheme encode 18 bits.  base64 encoding works on 6-bit groupings so: `C -> 0b000010, A -> 0b000000, A -> 0b000000` which is two bytes + two bits when concatenated together.  In the python example above we get back `b'\x08\x00'` -> `'0b00001000 0b00000000'` which is two bytes (16 bits) in hexidecimal notation.  The canonical decoding by the library is stripping the last two bits per the RFC.  Implementers should thus use a library capable of getting the index of the b64 characters according to the scheme (for this version string only) and not those written to give canonical decodings.
```
Comment (m00sey) [inline spec/spec.md:1232]: However, there are three

Comment (daidoji) [reply spec/spec.md:1232]: Three?

**Comment by daidoji** [comment]
Fixed parenthesis and made it a sentence.

File: spec/spec.md (hunk 1 above)
Comment (m00sey) [reply spec/spec.md:1242]: suggested change:

> However, there are three base64 characters in this version scheme encode 18 bits.

**Comment by daidoji** [comment]
sorry, I have EasyCLA but my PR attempt patch from the github UI failed for some reason.  Will try and recover tonight.

**Pull request closed without merging**